src/applications/disability-benefits/526EZ/helpers.jsx

```diff
@@ -606,18 +609,26 @@ export const noFDCWarning = (
 );
 
 
-const options = [
-  { id: 1, label: 'first' },
-  { id: 2, label: 'second' },
-  { id: 3, label: 'third' },
-  { id: 4, label: 'fourth' },
-];
+export function queryForFacilities(input = '') {
+  // Only search if the input has a length >= 3, otherwise, return an empty array
+  if (input.length < 3) {
+    return Promise.resolve([]);
+  }
 
+  const url = appendQuery('/facilities/suggested', {
+    type: ['health', 'dod_health'],
+    name_part: input // eslint-disable-line camelcase
+  });
 
-// Mimic querying the api for options
-export function queryForFacilities(input = '') {
-  // Emulate a fast api call
-  return Promise.resolve(options.filter(o => o.label.includes(input)));
+  return apiRequest(url, {},
+    (response) => {
+      return response.data.map(f => ({ id: f.id, label: f.attributes.name }));
+    },
+    (error) => {
+      Raven.captureMessage('Error querying for facilities', { input, error });
```

Comment:
Should we return a data structure in the event of an error?

Reply:
Do you mean log something else to sentry or the `return []` on the next line?

Reply:
I'm mainly asking if maintainers will expect a data structure to be returned from a failed request, it looks we typically use these to dispatch actions/logs, but perhaps I'm not considering the full range of how we use error handlers.

Reply:
Yeah, if we don't return anything from this function, when we get a failed query, it'll "pass" `undefined` to the Autosuggest field which I imagine will break things because it expects an array. I think we don't want to break the functionality entirely since the user can still enter free text, so I opted to not throw an error. Dispatching an error action is interesting, but I don't know how we'd use it right now.

We could use some discovery on how we'd want to let the user know that we're having difficulty querying for the list of facilities at the moment. Might be worth writing a ticket for the backlog.

Do you want to do that, or shall I?

Reply:
Sorry the motivation of my question wasn't clear, rather than passing an empty array, could an empty array be used as a default parameter in the argument that uses this? I'm not as familiar with the downstream portion of this, so perhaps that isn't practical.

Reply:
Ohhhhhhh, okay. I think I'm following now.
```js
// AutosuggestField
const results = queryForResults(input) || [];
```
That's what you're getting at?

Yeah, that's probably a generally better approach. I think for now, I'll leave it like this is, but write up another ticket to make it so in the common library. (I'd make the change now, but with the form library code freeze, I'd have to do it again, would be likely to forget, etc.)

Good call. :+1: